engine/src/main/java/org/hibernate/validator/internal/util/TypeVariableBindings.java

```diff
@@ -69,7 +69,7 @@ else if ( genericSuperType instanceof ParameterizedType ) {
 
 				boolean typeParameterFoundInSubType = false;
 				for ( Entry<TypeVariable<?>, TypeVariable<?>> subTypeParameter : bindings.entrySet() ) {
-					if ( typeArgument.equals( subTypeParameter.getValue() ) ) {
+					if ( typeArgument == subTypeParameter.getValue() ) {
```

Comment:
I think so. You compare the type arguments of the same class so they should be identical.